streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -311,6 +317,22 @@ public void run() {
                 "Updating global state failed. You can restart KafkaStreams to recover from this error.",
                 recoverableException
             );
+        } catch (final Exception e) {
+            if (this.streamsUncaughtExceptionHandler == null) {
+                throw e;
+            }
+            if (Thread.getDefaultUncaughtExceptionHandler() != null && newHandler) {
+                log.error("Stream's new uncaught exception handler is set as well as the deprecated old handler." +
+                        "The old handler will be ignored as long as a new handler is set.");
+            } else {
+                throw e;
+            }
+            if (this.streamsUncaughtExceptionHandler.handle(e) != StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.SHUTDOWN_APPLICATION) {
+                throw e;
+            } else {
+                log.warn("Exception in global stream thread cause the application to attempt to shutdown." +
+                        " This action will succeed only if there is at least one StreamThread running on ths client");
+            }
```

Comment:
```suggestion
            if (this.streamsUncaughtExceptionHandler.handle(e) = StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.SHUTDOWN_APPLICATION) {
                log.warn("Exception in global stream thread cause the application to attempt to shutdown." +
                        " This action will succeed only if there is at least one StreamThread running on ths client");
            }
```

This looked a bit off...